dig.go

```diff
@@ -692,6 +692,36 @@ func isFieldOptional(f reflect.StructField) (bool, error) {
 	return optional, err
 }
 
+type group struct {
```

Comment:
nit: Placing this in dig.go doesn't seem right. I would prefer to place it
closer to where it's used or in its own place. Suggestions:

- Place in result.go because it's meant for use by resultGrouped. That it's
  also used in param.go can be ignored because that's only for validation.
- Place in a new group.go file, in which case we'll want to move the _groupTag
  const definition there too.